website/static/js/upload.js

```diff
@@ -1,54 +1,85 @@
+/*
+  The following function manages interactions
+  between Trail Tracker's UI and backend, namely,
+  it allows users to add trails. Before interacting
+  with TT's backend, a roundtrip is first made to
+  Cloudinary, where media files are uploaded and stored.
+*/
 async function addImage() {
-    const url = "/upload-cloudinary";
+    const cloudinary_url = "/upload-cloudinary";
     const fileList = document.querySelectorAll("[type=file]");
-  
-    //iterate over the image files
+
+    /*  - - - - - - - - - - - - - - - - - - - - - - - - -
+        STEP 1: Upload images to Cloudinary
+
+    */
+    // Get a url for each image (once uploaded to cloudinary)
+    const image_urls = []
+
+    // iterate over the image files
     for (let i = 0; i < fileList.length; i++) {
         const formData = new FormData();
-      
+
         let file = fileList[i];
         formData.append("file", file.files[0]);
-        //this is where we send the data to the /upload-cloudinary end point
-        let cloud_res = await fetch(url, {
+
+        // this is where we send the data to the /upload-cloudinary end point
+        let cloud_res = await fetch(cloudinary_url, {
             method: "POST",
             body: formData
         })
-        //we are waiting for cloudinary to return the image object 
+        //we are waiting for cloudinary to return the image object
         let cloud_res_json = await cloud_res.json();
-  
-        //this is where we store the image in the database
-        let flask_resp = await fetch('/upload-trail', {
-            method: "POST",
-            body: JSON.stringify({
```

Comment:
This is an example of a comment I have. I can ask you to change something before it is approved. We can go back and forth on it – until this thread is resolved, the pull request cannot be approved!